contrib/intel/jenkins/tests.py

```diff
@@ -688,58 +693,100 @@ def execute_cmd(self):
+        for path,items in categories.items():
+            filename = f'{test_root}/{path}/testlist'
+            print(filename)
+            for item in items:
+                self.update_testlists(filename, item)
+
+    def build_mpich(self):
+        if (os.path.exists(f'{self.mpichpath}/config.log') !=True):
+            print("configure mpich")
+            os.chdir(self.mpichpath)
+            configure_cmd = f'./configure \
+                            --prefix=/home/cstbuild/cloudbees_middlewares/mpich_temp \
+                            --with-libfabric={self.mpi.libfab_installpath}  \
+                            --disable-oshmem --disable-fortran --without-ch4-shmmods \
+                            --with-device=ch4:ofi --without-ze'
+            print(configure_cmd)
+            common.run_command(['./autogen.sh'])
+            common.run_command(shlex.split(configure_cmd))
+            common.run_command(['make','-j'])
+            common.run_command(['make','install'])
+            os.chdir(self.pwd)
 
     @property
     def execute_condn(self):
-        return (self.mpi_type == 'impi' or \
-               (self.mpi_type == 'mpich' and self.core_prov == 'verbs'))
-
-    def execute_cmd(self, testgroupname):
-        print("Running Tests: " + testgroupname)
-        tests = []
-        time = None
-        os.chdir(f'{self.mpichsuitepath}/{testgroupname}')
-        tests = self.testgroup(testgroupname)
-        for test in tests:
-            testname = test[0]
-            nprocs = test[1]
-            args = test[2:]
-            for item in args:
-               itemlist =  item.split('=')
-               if (itemlist[0] == 'timelimit'):
-                   time = itemlist[1]
-            self.set_options(nprocs, timeout=time)
-            testcmd = f'./{testname}'
-            outputcmd = shlex.split(self.mpi.env + self.mpi.cmd + testcmd + '\'')
-            common.run_command(outputcmd)
-        os.chdir(self.pwd)
-
+        return ((self.mpi_type == 'impi' or \
+                self.mpi_type == 'mpich') and \
+               (self.core_prov == 'verbs' or self.core_prov == 'tcp'))
+    def execute_cmd(self):
+        if (self.mpi_type == 'mpich'):
+            configure_cmd = './configure'
+            if (self.weekly):
+                print(f'Weekly {self.mpi_type} mpichsuite tests')
+                os.chdir(self.mpichsuitepath)
+                common.run_command(shlex.split(self.mpi.env + configure_cmd +'\''))
+                self.exclude_tests(self.mpichsuitepath, self.core_prov)
+                testcmd = 'make testing'
+                outputcmd = shlex.split(self.mpi.env +testcmd + '\'')
+                common.run_command(outputcmd)
+                os.chdir(self.pwd)
+            else:
+                print(f"PR {self.mpi_type} mpichsuite tests")
+                os.chdir(self.mpichsuitepath)
+                configure_cmd += f' --with-mpi={cloudbees_config.build_dir}/mpich_temp'
```

Comment:
this configure_cmd is not needed, the f string can be placed inside the function call
This f string should use " instead if ' because there are spaces in it